src/asmcnc/skavaUI/screen_home.py

```diff
@@ -376,10 +349,39 @@ def on_pre_enter(self):
                 self.gcode_preview_widget.draw_file_in_xy_plane([])
                 self.gcode_preview_widget.get_non_modal_gcode([])
             except:
-                print 'No G-code loaded.'
+                print('No G-code loaded.')
 
             self.gcode_summary_widget.hide_summary()
 
+        else:
+            # File label at the top
+            self.file_data_label.text = "[color=333333]" + self.jd.job_name + "[/color]"    
+            self.gcode_summary_widget.display_summary()
+
+            # Preview file as drawing
+            try: 
+                Clock.schedule_once(self.preview_job_file, 0.05)
+            except:
+                log('Unable to preview file')
+
+        # Check if job recovery (or job redo) is available
+        if self.jd.job_recovery_cancel_line != None:
+
+            # Cancel on line -1 represents last job completing successfully
+            if self.jd.job_recovery_cancel_line == -1:
+                self.job_recovery_button_image.source = "./asmcnc/skavaUI/img/recover_job_disabled.png"
+
+            else:
+                self.job_recovery_button_image.source = "./asmcnc/skavaUI/img/recover_job.png"
+
+            # Line -1 being selected represents no selected line
+            if self.jd.job_recovery_selected_line == -1:
+                if self.jd.job_recovery_from_beginning:
```

Comment:
what does this flag do that the == -1 check doesn't do?

Reply:
-1 represents no selected line but not necessarily restarting from the beginning (e.g. no file is loaded, so no line is selected but not restarting from beginning) so extra flag needed to differentiate the cases